src/app/components/link-editor-popup/index.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useCallback, SyntheticEvent } from 'react';
+import ReactDOM from 'react-dom';
+import { TextField, IconButton, InputAdornment, Popover, ThemeProvider } from '@material-ui/core';
+import { EditorView, NodeView } from 'prosemirror-view';
+import { Node as ProsemirrorNode } from 'prosemirror-model';
+
+import LaunchIcon from '@material-ui/icons/Launch';
+import CloseIcon from '@material-ui/icons/Close';
+
+import { LINK_EDITOR_MAX_WIDTH, useLinkEditorStyles } from 'app/components/link-editor-popup/styles';
+import { ActionButton } from 'app/components/action-button';
+import { theme } from 'app/styles/theme';
+
+interface LinkEditorPopupProps {
+  editorView: EditorView | undefined;
+  onClose: (href?: string) => void;
+  node?: ProsemirrorNode;
+  y: number;
+  x: number;
+}
+
+export const LinkEditorPopup: React.FC<LinkEditorPopupProps> = ({ editorView, node, onClose, x, y }) => {
+  const classes = useLinkEditorStyles();
+  const [link, setLink] = useState<string>(node ? node.attrs.href : '');
+
+  const handleOnChange = useCallback(
+    (event: SyntheticEvent) => {
+      setLink(event.target['value']);
+    },
+    [setLink]
+  );
+
+  const handleLaunchClick = useCallback(() => {
+    window.open(link, '_blank');
+  }, [link]);
+
+  const handleClearField = useCallback(() => setLink(''), [setLink]);
+
+  const handleApplyClick = useCallback(() => {
+    editorView.focus();
+    onClose(link);
+  }, [editorView, link, onClose]);
+
+  const handleOnClose = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  if (!editorView) {
+    return <></>;
+  }
+
+  return (
+    <Popover
+      open={true}
+      onClose={handleOnClose}
+      anchorReference="anchorPosition"
+      anchorPosition={{ top: y, left: x }}
+      anchorOrigin={{
+        vertical: 'top',
+        horizontal: 'left'
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'left'
+      }}
+    >
+      <div className={classes.container}>
+        <TextField
```

Comment:
Annoyingly when i focus this element, the selection in the bio looses focus. Might be out of scope of this issue to fix, but wonder if [this](https://discuss.prosemirror.net/t/keeping-selection-while-using-the-menu-or-clicking-outside-the-document/578) might offer a solution of sorts?

Reply:
Yeah it's one of those things I haven't figured out yet. I checked the link, the proposed solution looks ok and it might work for us. preventingDefault works for buttons in the toolbar I'm afraid.